src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVMKernels.hpp

```diff
@@ -120,69 +145,830 @@ struct FluxKernel
+   * @param[in] dt time step size
+   * @param[inout] localMatrix the local CRS matrix
+   * @param[inout] localRhs the local right-hand side vector
+   */
+  FaceBasedAssemblyKernel( localIndex const numPhases,
+                           globalIndex const rankOffset,
+                           string const & dofKey,
+                           integer const capPressureFlag,
+                           string const & solverName,
+                           ElementRegionManager const & elemManager,
+                           STENCILWRAPPER const & stencilWrapper,
+                           arrayView1d< string const > const & targetRegionNames,
+                           arrayView1d< string const > const & fluidModelNames,
+                           arrayView1d< string const > const & relPermModelNames,
+                           arrayView1d< string const > const & capPresModelNames,
+                           arrayView1d< string const > const & permeabilityModelNames,
+                           real64 const & dt,
+                           CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                           arrayView1d< real64 > const & localRhs )
+    : m_numPhases( numPhases ),
+    m_rankOffset( rankOffset ),
+    m_capPressureFlag( capPressureFlag ),
+    m_dt( dt ),
+    m_stencilWrapper( stencilWrapper ),
+    m_seri( stencilWrapper.getElementRegionIndices() ),
+    m_sesri( stencilWrapper.getElementSubRegionIndices() ),
+    m_sei( stencilWrapper.getElementIndices() ),
+    m_localMatrix( localMatrix ),
+    m_localRhs( localRhs )
+  {
+    // unused for now, but may be useful if someone needs to work directly with relperms
+    GEOSX_UNUSED_VAR( relPermModelNames );
+
+    // dof numbers and ghost rank
+    {
+      m_ghostRankAccessor =
+        elemManager.constructArrayViewAccessor< integer, 1 >( ObjectManagerBase::viewKeyStruct::ghostRankString() );
+      m_ghostRankAccessor.setName( solverName + "/accessors/" + ObjectManagerBase::viewKeyStruct::ghostRankString() );
+      m_ghostRank = m_ghostRankAccessor.toNestedViewConst();
+
+      m_dofNumberAccessor = elemManager.constructArrayViewAccessor< globalIndex, 1 >( dofKey );
+      m_dofNumberAccessor.setName( solverName + "/accessors/" + dofKey );
+      m_dofNumber = m_dofNumberAccessor.toNestedViewConst();
+    }
+
+    // compflow
+    {
+      using keys = CompositionalMultiphaseBase::viewKeyStruct;
+      using namespace compflow;
+
+      m_gravCoefAccessor = elemManager.constructArrayViewAccessor< real64, 1 >( keys::gravityCoefString() );
+      m_gravCoefAccessor.setName( solverName + "/accessors/" + keys::gravityCoefString() );
+      m_gravCoef = m_gravCoefAccessor.toNestedViewConst();
+
+      m_presAccessor = elemManager.constructArrayViewAccessor< real64, 1 >( keys::pressureString() );
+      m_presAccessor.setName( solverName + "/accessors/" + keys::pressureString() );
+      m_pres = m_presAccessor.toNestedViewConst();
```

Comment:
@klevzoff @CusiniM @rrsettgast I have a question about the `elementViewAccessors` in the new kernel interface. 

My initial plan was to move the `elementViewAccessors` from the solver (ultimately removing `CompositionalMultiphaseBase::resetViews`) to the kernel, because I wanted to let the kernel choose which `elementViewAccessors` it needs. This is what is done in the current version of the PR, in which both the `elementViewAccessors` and the corresponding views are created in the constructor of the kernel interface (see highlighted code).

This plan works well for the CPU, but does not work on the GPU because the elementViewAccessors don't go to the GPU and make the code crash if they are stored in the kernel interface. Is there any way to keep this organization, but make it work on the GPU? Maybe the kernel factory can create the `elementViewAccessors`?

Alternatively, I can just leave the `elementViewAccessors` in the solver, and pass the views to the kernel factory, which will solve the problem on the GPU.